wagtail/core/models.py

```diff
@@ -4220,3 +4237,83 @@ def __str__(self):
     @cached_property
     def object_id(self):
         return self.page_id
+
+
+class Comment(models.Model):
+    """
+    A comment on a field, or a field within a streamfield block. This model stores the comment data that applies to all page revisions.
+    Any data which applies only for a single revision, or may change between revisions (such as position within a field) is stored on
+    CommentPosition.
+    """
+    page = models.ForeignKey(Page, on_delete=models.CASCADE, related_name='comments')
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name='comments')
+    text = models.TextField()
+
+    contentpath = models.TextField()
+    # This stores the field or field within a streamfield block that the comment is applied on, in the form: 'field', or 'field.block_id.field'
+    # This must be unchanging across all revisions, so we will not support (current-format) ListBlock or the contents of InlinePanels initially.
+    # When these are included, it should probably be in the form of a "changable contentpath" extension within CommentPosition
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    revision_created = models.ForeignKey(PageRevision, on_delete=models.CASCADE, related_name='created_comments')
+    # Comments are only shown on revisions after revision_created
+    revision_resolved = models.ForeignKey(PageRevision, on_delete=models.SET_NULL, related_name='resolved_comments', null=True, blank=True)
+    # A null value here indicates the comment is unresolved. Resolved comments can only be seen on revisions prior to their resolved revision.
+    # In most cases, revisions will be purged oldest-first, so deleting the comment when the revision is deleted is the correct behaviour as the resolved
+    # comment is now inaccessible. However, in cases where the deleted revision_resolved is not the oldest revision, the revision_resolved needs to be
+    # changed instead. This is done in PageRevision.delete()
+
+    class Meta:
+        verbose_name = _('comment')
+        verbose_name_plural = _('comments')
+
+    def __str__(self):
+        return _("Comment on Page '{0}', left by {1}: '{2}'").format(self.page, self.user, self.text)
```

Comment:
I'd suggest not marking these strings as translatable, since they're presumably not going to appear in user-facing UI and are just creating extra work for translators. (If we do keep them translatable, they should use old-style `%` string formatting so that Transifex / makemessages can catch any discrepancies between placeholders in the original and translated strings.)